tests/cli.rs

```diff
@@ -0,0 +1,278 @@
+use std::process::Command;
+
+use assert_cmd::prelude::*;
+use predicates::prelude::*;
+
+#[cfg(test)]
+mod usage {
+  use super::*;
+  #[test]
+  fn empty_args() -> Result<(), Box<dyn std::error::Error>> {
+    let mut cmd = Command::cargo_bin("tailcall")?;
+
+    cmd.arg("");
+    cmd
+      .assert()
+      .failure()
+      .stderr(predicate::str::contains("Usage: tailcall"));
+
+    Ok(())
+  }
+}
+
+// Check command tests
+#[cfg(test)]
+mod check {
+  use super::*;
+  #[test]
+  fn check_command_file_not_specified() -> Result<(), Box<dyn std::error::Error>> {
+    let mut cmd = Command::cargo_bin("tailcall")?;
+
+    cmd.arg("check");
+    cmd.assert().failure().stderr(predicate::str::contains(
+      "error: the following required arguments were not provided",
+    ));
+    cmd
+      .assert()
+      .failure()
+      .stderr(predicate::str::contains("Usage: tailcall check <FILE_PATH>"));
+
+    Ok(())
+  }
+
+  #[test]
+  fn check_command_file_doesnt_exist() -> Result<(), Box<dyn std::error::Error>> {
+    let mut cmd = Command::cargo_bin("tailcall")?;
+
+    cmd.arg("check").arg("test.file.doesnt.exist.graphql");
+    cmd
+      .assert()
+      .failure()
+      .stderr(predicate::str::contains("Error: No such file or directory"));
+
+    Ok(())
+  }
+
+  #[test]
+  fn check_command_file_exists_and_invalid() -> Result<(), Box<dyn std::error::Error>> {
+    let mut cmd = Command::cargo_bin("tailcall")?;
+
+    cmd
+      .arg("check")
+      .arg("tests/graphql/errors/test-const-with-inline.graphql");
+    cmd
+      .assert()
+      .failure()
+      .stderr(predicate::str::contains("Error: Validation Error"));
+
+    Ok(())
+  }
+
+  #[test]
+  fn check_command_file_exists_and_valid() -> Result<(), Box<dyn std::error::Error>> {
+    let mut cmd = Command::cargo_bin("tailcall")?;
+
+    cmd
+      .arg("check")
+      .arg("examples/jsonplaceholder.graphql")
+      .arg("--n-plus-one-queries")
+      .arg("--schema");
```

Comment:
should we also check that the cli outputs schema in that case? at least part of it

Reply:
I'll see if this is possible.